src/ResponseSerializer.php

```diff
@@ -3,23 +3,41 @@
 namespace Spatie\ResponseCache;
 
 use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\BinaryFileResponse;
 
 class ResponseSerializer
 {
+    const RESPONSE_TYPE_NORMAL = 1;
+    const RESPONSE_TYPE_FILE = 2;
+
     public function serialize(Response $response): string
     {
-        $content = $response->getContent();
+        $type = self::RESPONSE_TYPE_NORMAL;
+
+        if ($response instanceof BinaryFileResponse) {
+            $content = $response->getFile()->getPathname();
+            $type = self::RESPONSE_TYPE_FILE;
+        } else {
```

Comment:
Generally speaking I very much dislike else statement. Extract this code to it's own function where you can use early returns.